src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -47,19 +47,33 @@ This is a sphere with inner radius 1 and outer radius 2.
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP 
+    θ::TT 
+
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+        
+end
+
+# function Ellipsoid{T,CO}(r::TR,φ::TP,θ::TT,origin::CartesianPoint{T}=zero(CartesianPoint{T}),
+#     rotation::SMatrix{3,3,T,9}=one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing} 
+#     Ellipsoid{T,CO,TR,TP,TT}(r,φ,θ,origin,rotation)
+# end
+
+function Ellipsoid(::Type{T}=Float64, ::Type{CO}=ClosedPrimitive; r::TR = 1.0, origin = zero(CartesianPoint{T}), 
+    rotation = one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T}
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    TP=Nothing
+    TT=Nothing
+    return Ellipsoid{T,CO,TR,TP,TT}(r, nothing, nothing, origin, rotation)
```

Comment:
Please put `Nothing` directly in the constructor at the end to keep it small. 

Hmm, I am still unsure whether it would be better here if we put `T` and `CO` into 
the curly brackets: `Ellipsoid{T, CO}(...`
And then add methods which call this method to add defaults. 
E.g. `Ellipsoid(::Type{T} = Float64, ::Type{CO} = ClosedPrimitve; kwargs...) = Ellipsoid{T, CO}(; kwargs...)`